frontend/src/components/elements/PlotlyChart/CustomTheme.tsx

```diff
@@ -0,0 +1,640 @@
+      gridcolor: getGray70(theme),
+      bgcolor: colors.bgColor,
+      title: {
+        font: {
+          family: genericFonts.bodyFont,
+          size: fontSizes.smPx,
+        },
+      },
+      color: getGray70(theme),
+      aaxis: {
+        gridcolor: getGray70(theme),
+        linecolor: getGray70(theme),
+        tickfont: {
+          family: genericFonts.bodyFont,
+          size: fontSizes.twoSmPx,
+        },
+      },
+      baxis: {
+        linecolor: getGray70(theme),
+        gridcolor: getGray70(theme),
+        tickfont: {
+          family: genericFonts.bodyFont,
+          size: fontSizes.twoSmPx,
+        },
+      },
+      caxis: {
+        linecolor: getGray70(theme),
+        gridcolor: getGray70(theme),
+        tickfont: {
+          family: genericFonts.bodyFont,
+          size: fontSizes.twoSmPx,
+        },
+      },
+    },
+  }
+
+  assign(layout, streamlitTheme)
+}
+
+/**
+ * Applies the colorscale, colors for unique graphs, and discrete coloring
+ * for in general through assigning properties in spec.data
+ * @param data - spec.data
+ */
+export function applyStreamlitThemeData(data: any): void {
+  const { colors } = useTheme()
+  applyColorscale(data)
+  applyUniqueGraphColorsData(data)
+  changeDiscreteColors(data)
+  data.forEach((entry: any) => {
+    if (entry.marker !== undefined) {
+      entry.marker.line = assign(entry.marker.line, {
+        width: 0,
+        color: colors.transparent,
+      })
+    }
+  })
+}
+
+/**
+ * This applies specific changes for spec.layout.template.data because
+ * spec.data does not fully catch these and neither does spec.layout.template.layout
+ * @param data - spec.layout.template.data
+ * @param theme - Theme from useTheme()
+ */
+export function applyStreamlitThemeTemplateData(
+  data: any,
+  theme: Theme
+): void {
+  if (data !== undefined) {
+    data.bar.forEach((entry: any) => {
+      if (entry.marker !== undefined && entry.marker.line !== undefined) {
+        entry.marker.line = assign(entry.marker.line, {
+          color: getGray30(theme),
+        })
+      }
+    })
+    const graphs = Object.values(data)
+    graphs.forEach((graph: any) => {
+      if (graph.colorbar !== undefined && graph.colorbar.ticks === "") {
+        // make tick values show
+        delete graph.colorbar.ticks
+      }
+    })
+  }
+}
+
+/**
+ * Applies the Streamlit theme by overriding properties in
+ * spec.data, spec.layout.template.data, and spec.layout.template.layout
+ * @param spec - spec
+ */
+export function applyStreamlitTheme(spec: any): void {
+  const theme: Theme = useTheme()
+  applyStreamlitThemeTemplateLayout(spec.layout.template.layout, theme)
+  applyStreamlitThemeTemplateData(spec.layout.template.data, theme)
+  applyStreamlitThemeData(spec.data)
+  if ("title" in spec.layout) {
+    spec.layout.title = assign({
+      text: `<b>${spec.layout.title.text}</b>`,
+    })
+  }
+}
+
+export function applyThemeDefaults(config: any, theme: Theme): any {
```

Comment:
I assume this method is still based on the vega lite state? I think you still need to migrate the logic form `layoutWithThemeDefaults` to here:

https://github.com/streamlit/streamlit/blob/a42987209e2f8bdc0eacb22375ffa2676971ae0e/frontend/src/components/elements/PlotlyChart/PlotlyChart.tsx#L90